distro_test.go

```diff
@@ -163,6 +177,23 @@ configuration: error in GetConfiguration: failed to convert %q to UTF16
 	}
 }
 
+func TestGUID(t *testing.T) {
+	// This test validates that the GUID is properly printed.
+	// Note that windows.GUID has a String method printing the expected
+	// format "{XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX}", but syscall.GUID
+	// does not have such method and prints its contents like any other
+	// struct.
+	distro := wsl.NewDistro(uniqueDistroName(t))
+	err := distro.Register(emptyRootFs)
+	require.NoError(t, err, "could not register empty distro")
+
+	guid, err := distro.GUID()
+	require.NoError(t, err, "could not obtain GUID")
+
+	pattern := regexp.MustCompile(`^\{[A-F0-9]{8}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{4}-[A-F0-9]{12}\}$`)
```

Comment:
that pattern would be a nice utility pattern in general for validating Windows UIs, but here, you are only using it once, so ok :)

Reply:
Yeah, in the end I trusted the Win32 API to do this for me everywhere else. There's also Google's implementation that does not depend on a syscall: https://pkg.go.dev/github.com/google/uuid#Parse